src/main/java/edu/wpi/grip/core/util/ErrorWitness.java

```diff
@@ -0,0 +1,61 @@
+package edu.wpi.grip.core.util;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.events.ErrorClearedEvent;
+import edu.wpi.grip.core.events.ErrorEvent;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Witness and reports errors.
+ * {@link #flagError} should be used to flags the witness that an flagError has has occurred.
+ * This will post an {@link ErrorEvent} to the {@link EventBus}
+ */
+public final class ErrorWitness {
+    private final EventBus eventBus;
+    private final Object origin;
+    private final AtomicBoolean isErrorState = new AtomicBoolean(false);
+
+    public ErrorWitness(final EventBus eventBus, final Object origin) {
+        this.eventBus = checkNotNull(eventBus, "The event bus can not be null");
+        this.origin = checkNotNull(origin, "The origin can not be null");
+    }
+
+    /**
+     * Indicates to the witness that an flagError has occurred. This will also post an {@link ErrorEvent} to the {@link EventBus}
+     *
+     * @param throwable The flagError that this is reporting
+     * @param message   Any additional details that should be associated with this message.
+     */
+    public final void flagError(final Throwable throwable, final String message) {
```

Comment:
IMO it's not super important what it's called if it's obvious what it does. So if you write more of a header I'm OK with leaving it as it is.